test/runtime/configMatters/comm/atomics.suppressif

```diff
@@ -1,17 +1,22 @@
-#! /usr/bin/env bash
+#! /usr/bin/env python
 #
-# CCE versions greater than 8.6.4 have issues with processor-local atomics,
-# so suppress this case for now.
+# CCE 8.6.4 introduced a bug with __sync_fetch_and_sub that was only fixed in
+# 8.7.4, so suppress this case for those versions
 #
-eval `$CHPL_HOME/util/printchplenv --make`
 
-if [ "$CHPL_MAKE_TARGET_COMPILER" = cray-prgenv-cray -a "$CHPL_MAKE_COMM" = none ]; then
+import os, sys
 
-  cce_version=`module list -t 2>&1 | grep "^cce/" | sed -e "s,^cce/,,"`
+def_home = os.path.join(os.path.dirname(__file__), '..', '..', '..', '..')
+chpl_home = os.path.abspath(os.getenv('CHPL_HOME', def_home))
+chplenv_dir = os.path.join(chpl_home, 'util', 'chplenv')
+sys.path.insert(0, chplenv_dir)
 
-  if [ -n "$cce_version" -a "$cce_version" \> "8.6.4" ]; then
-    echo 1
-    exit 0
-  fi
-fi
-echo 0
+from compiler_utils import CompVersion, get_compiler_version
+import chpl_compiler, chpl_comm
+
+comm = chpl_comm.get()
+compiler = chpl_compiler.get('target')
+cce = compiler == 'cray-prgenv-cray'
+buggy_cce = cce and CompVersion('8.6.4') <= get_compiler_version(compiler) < CompVersion('8.7.4')
+
+print(comm == 'none' and buggy_cce)
```

Comment:
Do we also want to check `CHPL_NETWORK_ATOMICS` here?  I.e., is there a problem even when `comm!='none'`, if we're not using network atomics?

Reply:
Oh, that's a good catch. Yes, we should be checking for `CHPL_NETWORK_ATOMICS == 'none'` instead of `comm == none`.  This test will also fail under ugni with network_atomics=none and it fails under gasnet. We didn't notice this before since we don't test gasnet an ARM (and ARM is place we're using new enough compiler to hit this) and we always run ugni with network atomics.

Reply:
(and note that we won't hit this for `CHPL_NETWORK_ATOMICS=ugni` even when we use processor atomics in the comm API (-nl1 or atomic not registered) because https://github.com/chapel-lang/chapel/issues/11791 means we never call __sync_fetch_and_sub)